models/inherit_stock.py

```diff
@@ -49,63 +76,65 @@ def unlink(self):
 
     @api.model
     def create(self,vals):
-	record = super(stock_warehouse, self).create(vals)
-	values = {}
-	if record.shop_id.public:
-	    values['company_type'] = 'company'
-	    values['is_company'] = True
-	    values['category_id'] = [(6,0,[self.env.ref('website_shops_map.partner_shop_category_id').id])]
-	record.shop_id.write(values)
-	return record
-
-    @api.one
+        record = super(stock_warehouse, self).create(vals)
+        values = {}
+        if record.shop_id.public:
+            values['company_type'] = 'company'
+            values['is_company'] = True
+            values['category_id'] = [(6,0,[self.env.ref('website_shops_map.partner_shop_category_id').id])]
+        record.shop_id.write(values)
+        return record
+
     def write(self,vals):
-	record = super(stock_warehouse, self).write(vals)
-	values = {'name': self.name}
-	#if warehouse is shop
-	if self.shop_id.public:
-	    #if there are no coordinates - count them
-	    if not self.shop_id.partner_latitude:
-		coordinates = self._get_lat_lng()
-		values['partner_latitude'] = coordinates[0]
-		values['partner_longitude'] = coordinates[1]
-		values['date_localization'] = datetime.today().strftime('%Y-%m-%d')
-	    # assign first shop photo as shop partner avatar image
-	    if self.shop_id.image_ids:
-		values['image'] = self.shop_id.image_ids[0].image_medium
-	    else:
-		# assign company logo as shop partner avatar image
-		if self.partner_id:
-		    values['image'] = self.env['res.company'].browse(self.partner_id.id).logo
-	self.shop_id.write(values)
-	return record
-
-
-    @api.multi
+        record = super(stock_warehouse, self).write(vals)
+        values = {'name': self.name}
+        # if warehouse is shop
+        if self.shop_id.public:
```

Comment:
1. Having a variable for `self.shop_id` would be nice
2. Write can be called on recordset (more than one record), in such case `self.shop_id.public` will raise singleton error.